src/use-swr.ts

```diff
@@ -330,7 +331,13 @@ function useSWR<Data = any, Error = any>(
 
   // check if component is mounted in suspense mode
   const initialMountedRef = useRef(false)
-
+  if (initialMountedRef.current && keyRef.current !== key) {
```

Comment:
I feel this change is trying to approach the similar functionality of the return values? wonder if it's possible to unify them together in one place to cover these corner cases

```js
error: {
  // `key` might be changed in the upcoming hook re-render,
  // but the previous state will stay
  // so we need to match the latest key and data (fallback to `initialData`)
  get: function() {
    stateDependencies.current.error = true
    return keyRef.current === key ? stateRef.current.error : initialError
  },
  enumerable: true
},
data: {
  get: function() {
    stateDependencies.current.data = true
    return keyRef.current === key ? stateRef.current.data : initialData
  },
  enumerable: true
},
isValidating: {
  get: function() {
    stateDependencies.current.isValidating = true
    return key ? stateRef.current.isValidating : false
  },
  enumerable: true
}
```